crud/ProductManager/ProductManager/Classes/Manager.cs

```diff
@@ -0,0 +1,273 @@
+                            Name = (string)reader["Name"],
+                            Description = (string)reader["Description"],
+                            Weight = (double)reader["Weight"],
+                            Width = (double)reader["Width"],
+                            Height = (double)reader["Height"],
+                            Length = (double)reader["Length"]
+                        };
+
+                        _products.Add(product);
+                    }
+                }
+                
+                connection.Close();
+                return _products;
+            }
+        }
+
+        public List<Order> GetOrdersByFilter(int filterYear, int filterMonth, string filterStatus, int filterProductId)
+        {
+            var filteredOrders = _orders;
+
+            using (var connection = new NpgsqlConnection(_connectionString))
+            {
+                connection.Open();
+                using (var command = new NpgsqlCommand("GetFilteredOrders", connection))
+                {
+                    command.CommandType = System.Data.CommandType.StoredProcedure;
+
+                    command.Parameters.AddWithValue("filter_year", filterYear);
+                    command.Parameters.AddWithValue("filter_month", filterMonth);
+                    command.Parameters.AddWithValue("filter_status", filterStatus);
+                    command.Parameters.AddWithValue("filter_product_id", filterProductId);
+
+                    command.Parameters.Add(new NpgsqlParameter("result_set", NpgsqlTypes.NpgsqlDbType.Refcursor)
+                    {
+                        Direction = System.Data.ParameterDirection.Output
+                    });
+                    using (var reader = command.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            var order = new Order
+                            {
+                                Id = reader.GetInt32(0),
+                                CreateDate = reader.GetDateTime(1),
+                                UpdatedDate = reader.GetDateTime(2),
+                                ProductId = reader.GetInt32(3),
+                                Status = reader.GetInt16(4)
+                            };
+                            filteredOrders.Add(order);
+                        }
+                    }
+
+                    return filteredOrders;
+                }
+            }
+        }
+
+        public void BulkDeleteOrders(List<int> orderIds)
+        {
+            if (orderIds == null || orderIds.Count == 0)
+            {
+                return;
+            }
+
+            using (var connection = new NpgsqlConnection(_connectionString))
+            {
+                connection.Open();
+
+                var cmd = connection.CreateCommand();
+                cmd.CommandText = "DELETE FROM public.\"Order\" WHERE \"Id\" = ANY(@orderIds)";
```

Comment:
never seen such construction 'WHERE "Id" = ANY(@orderIds)' , I would use 'WHERE Id IN (@orderIds)', but if it works, than fine, leave it like that

Reply:
add filters to the delete, same way you did on update